// @atlaskit/avatar/dist/cjs/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ICON_SIZES = exports.ICON_OFFSET = exports.CSS_VAR_AVATAR_BGCOLOR = exports.BORDER_WIDTH = exports.AVATAR_SIZES = exports.AVATAR_RADIUS = exports.ACTIVE_SCALE_FACTOR = void 0;
var _constants = require("@atlaskit/theme/constants");
// eslint-disable-next-line @atlaskit/design-system/no-deprecated-imports

var AVATAR_SIZES = exports.AVATAR_SIZES = {
  xsmall: (0, _constants.gridSize)() * 2,
  small: (0, _constants.gridSize)() * 3,
  medium: (0, _constants.gridSize)() * 4,
  large: (0, _constants.gridSize)() * 5,
  xlarge: (0, _constants.gridSize)() * 12,
  xxlarge: (0, _constants.gridSize)() * 16
};

// border radius only applies to "square" avatars
var AVATAR_RADIUS = exports.AVATAR_RADIUS = {
  xsmall: 2,
  small: 2,
  medium: 3,
  large: 3,
  xlarge: 6,
  xxlarge: 12
};
var BORDER_WIDTH = exports.BORDER_WIDTH = 2;
var ACTIVE_SCALE_FACTOR = exports.ACTIVE_SCALE_FACTOR = 0.9;
var CSS_VAR_AVATAR_BGCOLOR = exports.CSS_VAR_AVATAR_BGCOLOR = '--avatar-background-color';
var ICON_SIZES = exports.ICON_SIZES = {
  small: 12,
  medium: 14,
  large: 15,
  xlarge: 18
};
var ICON_OFFSET = exports.ICON_OFFSET = {
  small: 0,
  medium: 0,
  large: 1,
  xlarge: 7
};
